backend/src/main/java/com/woowacourse/kkogkkog/coupon/domain/CouponState.java

```diff
@@ -0,0 +1,43 @@
+package com.woowacourse.kkogkkog.coupon.domain;
+
+import java.time.LocalDateTime;
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Embeddable
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@Getter
+public class CouponState {
+
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private CouponStatus couponStatus;
+
+    @Column
+    private LocalDateTime meetingDate;
+
+    public CouponState(CouponStatus couponStatus, LocalDateTime meetingDate) {
+        this.couponStatus = couponStatus;
+        this.meetingDate = meetingDate;
+    }
+
+    public static CouponState ofReady() {
+        return new CouponState(CouponStatus.READY, null);
+    }
+
+    public void changeStatus(CouponEvent couponEvent) {
+        this.couponStatus = couponStatus.handle(couponEvent.getType());
+        updateMeetingDate(couponEvent);
+    }
+
+    private void updateMeetingDate(CouponEvent couponEvent) {
+        if (couponEvent.shouldUpdateMeetingDate()) {
+            this.meetingDate = couponEvent.getMeetingDate();
+        }
+    }
```

Comment:
생각해보니 이렇게 해보는건 어떨까요? 
- 기존 구조: CouponEvent의 내부 상태로 모든 것을 관리함에 따라 요구사항이 변경되어도 CouponEvent만 수정함으로써 OCP를 충족하자!
- 새로운 구조: 그냥 CANCEL, DECLINE일 때 CouponEvent의 type 필드 값이 null이라는 점을 파악하는 대신 그냥 null을 명시적으로 할당하자!

```suggestion
    private void updateMeetingDate(CouponEvent couponEvent) {
        if (couponEvent.shouldAddMeetingDate()) {
            this.meetingDate = couponEvent.getMeetingDate();
        }
        if (couponEvent.shouldRemoveMeetingDate()) {
            this.meetingDate = null;
        }
    }
```
CouponEvent 쪽은 아래와 같이 수정하면 될 것 같아요.

```java
@Getter
public class CouponEvent {

    private final CouponEventType type;
    private final LocalDateTime meetingDate;

   // ...
   
    public boolean shouldAddMeetingDate() {
        return type == REQUEST;
    }

    public boolean shouldRemoveMeetingDate() {
        return type == CANCEL || type == DECLINE;
    }
}
```